### PR TITLE
rename package to `spago-legacy` to help NixPkgs

### PR DESCRIPTION
**Pull request by peterbecich**
### Description of the change

This package is currently `spago` in NixPkgs:
https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=spago

The Nix configuration for that is generated from this Cabal file. Renaming to `spago-legacy` will help clear the path for NixPkgs update to the new Spago.

I think it's unnecessary for the `spago` binary to be renamed to `spago-legacy`.

If this PR is merged, NixPkgs will be updated to the latest commit in this repo `spago-legacy`.

```
cabal build --enable-tests
```
succeeds with GHC 9.2

`stack build` succeeds.

Please merge this PR and make a fix version/release 0.21.1 or maybe even 0.21.0.1. This is probably necessary for Nix to grab the latest commit, I don't think it works directly off the branch.

### Checklist:

- [ ] Added the change to the "Unreleased" section of the changelog
- [ ] Added some example of the new feature to the `README`
- [x] Added a test for the contribution (if applicable)

**P.S.**: the above checks are not compulsory to get a change merged, so you may skip them. However, taking care of them will result in less work for the maintainers and will be much appreciated 😊

### CODE REVIEW

**Comment by f-f** [comment]
We don't have release CI set up on this repo, so it might take more work to build the binaries. Will nixpkgs need the binaries or will it build them off the branch? In other words: will just cutting the release tag enough for nixpkgs?

**Comment by peterbecich** [comment]
NixPkgs will not need the binaries. 

From what I can understand about this script
https://github.com/NixOS/nixpkgs/blob/9860009d87706af95eb94bc5369f86741e047135/pkgs/development/tools/purescript/spago-legacy/update.sh#L22-L31
the release bundle is necessary but you can omit the binaries. In fact, the release bundle can probably be empty. A tag without a release is insufficient. The script is going to grab the latest item in https://api.github.com/repos/purescript/spago-legacy/releases, parse the `tag` from that, and then `cabal2nix` will fetch that `tag`.

Thanks

**Comment by f-f** [comment]
Thanks, that makes sense. I can merge this PR and cut a new tag, but we'll need to get CI going first - the `macOS` build fails with the purescript installer failing `Check if a prebuilt 0.15.8 binary is provided for darwin-arm64`. Which is strange because the build is not running on an ARM machine.

**Comment by peterbecich** [comment]
I can disable the unnecessary jobs in `.github` in this PR, leaving only `build ubuntu-latest` and `Cabal build`, is that acceptable?

**Comment by peterbecich** [comment]
I don't think the Mac build is necessary to support the Darwin platform of the Nix build:

https://search.nixos.org/packages?channel=unstable&show=spago&from=0&size=50&sort=relevance&type=packages&query=spago
Platforms

    [aarch64-darwin](https://hydra.nixos.org/job/nixos/trunk-combined/nixpkgs.spago.aarch64-darwin)
    [aarch64-linux](https://hydra.nixos.org/job/nixos/trunk-combined/nixpkgs.spago.aarch64-linux)
    [i686-linux](https://hydra.nixos.org/job/nixos/trunk-combined/nixpkgs.spago.i686-linux)
    [x86_64-darwin](https://hydra.nixos.org/job/nixos/trunk-combined/nixpkgs.spago.x86_64-darwin)
    [x86_64-linux](https://hydra.nixos.org/job/nixos/trunk-combined/nixpkgs.spago.x86_64-linux)
    
It is going to build Spago directly from source, given the release package.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
       - uses: actions/checkout@v2
       - name: Setup Haskell (macOS and Windows)
         if: runner.os != 'Linux'
-        uses: haskell/actions/setup@v1
+        uses: haskell-actions/setup@v2
```
Comment (peterbecich) [inline .github/workflows/build.yml:41]: https://github.com/haskell-actions/setup

**Comment by peterbecich** [comment]
@f-f please review again

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -39,8 +39,8 @@ jobs:
         include:
```
Comment (f-f) [inline .github/workflows/release.yml]: We should entirely remove the `release` CI file - it's the one that packages things on release, but we won't be doing that anymore

File: spago-legacy.cabal [hunk 3]
```diff
@@ -1,10 +1,10 @@
 cabal-version: 2.4
 
-name:           spago
+name:           spago-legacy
 version:        0.21.0
```
Comment (f-f) [inline spago-legacy.cabal]: I guess we should bump this to `0.21.1`?

**Comment by peterbecich** [comment]
@f-f please review again

**Review by f-f — APPROVED** [review]

**Comment by f-f** [comment]
I merged and cut the new release. Please keep me posted on the progress in nixpkgs!

> I think it's unnecessary for the `spago` binary to be renamed to `spago-legacy`.

Let's do the renaming, it's quite confusing to have the two things share the same binary name.

**Comment by peterbecich** [comment]
Thanks @f-f , here is the WIP https://github.com/NixOS/nixpkgs/pull/324944

I will rename the binary to `spago-legacy`. This may require another PR and release, t.b.d.

**Pull request merged**